src/app-router.js

```diff
@@ -28,7 +28,11 @@ export class AppRouter extends Router {
   reset() {
     super.reset();
     this.maxInstructionCount = 10;
-    this._queue = [];
+    if (!this._queue) {
+      this._queue = [];
```

Comment:
I think the inline assignment could be problematic. It depends on the code-gen. If the super constructor runs before the inline initialization, then I think you could get an error because the super constructor calls reset and that would attempt to set the length before the queue was initialized. Does that make sense?